src/v/redpanda/admin_server.cc

```diff
@@ -3339,3 +3346,52 @@ void admin_server::register_shadow_indexing_routes() {
           return sync_local_state_handler(std::move(req));
       });
 }
+
+ss::future<> admin_server::restart_redpanda_service(service_kind service) {
+    if (service == service_kind::schema_registry) {
+        // Checks specific to schema registry
+        if (_schema_registry == nullptr) {
+            throw ss::httpd::server_error_exception(
+              "Schema Registry is undefined. Is it set in the .yaml config "
+              "file?");
+        }
+
+        try {
+            co_await _schema_registry->restart();
+        } catch (...) {
+            throw ss::httpd::server_error_exception(
+              "Unknown issue restarting schema_registry");
+        }
+    } else {
+        throw ss::httpd::server_error_exception(
+          fmt::format("Restart is unsupported for service: {}", service));
+    }
+}
+
+ss::future<ss::json::json_return_type>
+admin_server::redpanda_services_restart_handler(
+  std::unique_ptr<ss::httpd::request> req) {
+    auto service_param = req->get_query_param("service");
+    if (service_param.empty()) {
+        throw ss::httpd::not_found_exception("Service name is required");
+    }
```

Comment:
Is this check needed? I thought required params already throw a bad_request (it's tested for) rather than a not_found?

Reply:
Ah good point, bad_request is thrown before control gets here.